pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -30,6 +31,10 @@ let gnupg-minimal = gnupg.override {
   zlib = null;
   bzip2 = null;
 };
+
+# to avoid circular dependencies, build cryptsetup with a lvm2 without udev support
+_cryptsetup = cryptsetup.override { lvm2 = lvm2.override { udev = null; }; };
```

Comment:
Don't we usually do these in `all-packages.nix`? There is no real usage for `lvm2` in this package beyond this. It is some detail that we have to take care of because of how we wire things (Nix) and not how systemd is being built.

Reply:
hmmmh - not sure about this one. I personally prefer keeping these hacks inside the specific derivation that needs it, to reduce clutter in `all-packages.nix`. I'd like to keep it consitent with `gnupg-minimal` above.

Reply:
I did dig a bit deeper. `cryptsetup` really only uses the `devmapper` library which is shipped with `lvm2`. So `cryptsetup` in general is perfectly fine with a `lvm2` without udev.

This means, we're perfectly fine building `cryptsetup` defined in `all-packages.nix` this way, and I can remove the `_cryptsetup` here.

I also did some closure slimming by introducing multiple outputs for the `lvm2` derivation and disabling the `cmdlib` library, which only seems to be used to [simplify LV management tasks via a web interface](https://www.redhat.com/archives/linux-lvm/2008-September/msg00029.html) (?)